frontend/src/metabase/visualizations/visualizations/PieChart/PieChart.tsx

```diff
@@ -106,17 +106,19 @@ export function PieChart(props: VisualizationProps) {
       const label = s.data.name;
 
       // Hidden slices don't have a percentage
-      if (s.data.normalizedPercentage === 0) {
-        return label;
-      }
+      const sliceHidden = s.data.normalizedPercentage === 0;
+      const percentDisabled =
+        settings["pie.percent_visibility"] !== "legend" &&
+        settings["pie.percent_visibility"] !== "both";
 
-      const percent =
-        settings["pie.percent_visibility"] === "legend" ||
-        settings["pie.percent_visibility"] === "both"
-          ? formatters.formatPercent(s.data.normalizedPercentage, "legend")
-          : undefined;
```

Comment:
This was the crux of the issue. Because we returned `["label", undefined]` in this case, the below condition from LegendVertical.tsx would not run, causing the width of the legend to be too small.

``` typescript
if (
      sampleTitle &&
      previousSampleTitle &&
      previousSampleTitle.length !== sampleTitle.length
    ) {
      this.setState({ listWidth: null }, () => {
        const listWidth = this.listRef.current.getBoundingClientRect().width;
        this.setState({ listWidth });
      });
    }
```